unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -1775,6 +1786,75 @@ handleGist :: GistInput -> Cli ()
 handleGist (GistInput repo) =
   doPushRemoteBranch (GistyPush repo) Path.relativeEmpty' SyncMode.ShortCircuit
 
+handleDiffNamespaceToPatch :: Text -> DiffNamespaceToPatchInput -> Cli ()
+handleDiffNamespaceToPatch description input = do
+  Cli.Env {codebase} <- ask
+
+  branch1 <- Branch.head <$> Cli.resolveBranchId (input ^. #branchId1)
+  branch2 <- Branch.head <$> Cli.resolveBranchId (input ^. #branchId2)
+  branchDiff <- liftIO (BranchDiff.diff0 branch1 branch2)
+
+  -- Given {old referents} and {new referents}, create term edit patch entries as follows:
+  --
+  --   * If the {new referents} is a singleton set {new referent}, proceed. (Otherwise, the patch we might create would
+  --     not be a function, which is a bogus/conflicted patch).
+  --   * If the new referent is a term reference, not a data constructor, proceed. (Patches currently can't track
+  --     updates to data constructors).
+  --   * For each old term reference (again, throwing constructors away) in {old referents}, create a patch entry that
+  --     maps the old reference to the new. The patch entry includes the typing relationship between the terms, so we
+  --     look the references' types up in the codebase, too.
+  let termNamespaceUpdateToTermEdits :: (Set Referent, Set Referent) -> Sqlite.Transaction (Set (Reference, TermEdit))
+      termNamespaceUpdateToTermEdits (refs0, refs1) =
+        case Set.asSingleton refs1 of
+          Just (Referent.Ref ref1) ->
+            Codebase.getTypeOfTerm codebase ref1 >>= \case
+              Nothing -> pure Set.empty
```

Comment:
When is this case possible? Should we throw some sort of error here instead?

Reply:
If it's an indicator of some sort of transactionality failure and isn't expected to happen in normal circumstances I guess I'd prefer that it throw a bug report error so we get reports if it's acting up.

Reply:
I'm not sure I'd call it a bug, it's just that a term was deleted somehow between a previous query and this one. Either way, though, I don't think it's going to bite, so I'm inclined to just leave it as-is.